### PR TITLE
Refactor CLI Projects to Single Entrypoint

### PR DESCRIPTION
**Pull request by gfs**
This PR mainly refactors the many CLI projects to a single entrypoint. From a CLI perspective they now can be accessed with verbs against a the single `oss-gadget-cli` binary. This also standardizes calling patterns for each tool using the `BaseTool` class to a `RunAsync` method which takes the appropriate type of options. If the project previously had a `RunAsync` entrypoint with a different return type the old entrypoint was renamed `LegacyRunAsync`. This refactor does not attempt to change execution behavior, merely to simplify project structure and publishing, specifically to enable publishing the single CLI binary to Nuget withotu having to manage many project publications.

Also updates all dependencies.

Changes Model.User.Id field to long (from int) to match changes in OctoKit library which populates that field.

Marked as draft for feedback and refactor of build pipeline.

### CODE REVIEW

**Review by danfiedler-msft — APPROVED** [review]
Overall, these changes look good to me.

My personal preference would be to rename the `LegacyRunAsync` methods to reflect what the method does. The `RunAsync` method makes sense because it comes from the base class and must apply to all commands but there's no reason the `LegacyRunAsync` methods can't be renamed to be more specific. It isn't worth holding the PR up over.

**Comment by danfiedler-msft** [comment]
I think the different verbs/options `HelpText` should use the same text as the first Example.

For example, "Calculate a risk metric for the given package" provides more information to the user than "Run risk calculator tool"

Any help text that says, "runs the Foo tool" makes me ask "...but what does Foo tool _do_?"

File: src/oss-gadget-cli/Program.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+using Microsoft.CST.OpenSource.OssGadget.Tools.HealthTool;
+using System.Reflection;
+
+class OssGadgetCli : OSSGadget
+{
+    static async Task<int> Main(string[] args)
+    {
+        var cli = new OssGadgetCli();
+        await cli.ExecuteAsync(args);
+        return (int)cli._returnCode;
```
[the remark this answers is not in the document]
Comment (gfs) [reply src/oss-gadget-cli/Program.cs]: Good point. I've simplified this.

**Pull request merged**